compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1409,7 +1417,11 @@ public void visit(BLangOrderByVariable orderByVariable) {
     public void visit(BLangSelectExpression selectExpression) {
         ExpressionNode expressionNode = selectExpression.getExpression();
         if (!isSiddhiRuntimeEnabled) {
-            this.typeChecker.checkExpr((BLangExpression) expressionNode, env);
+            if (isGroupByAvailable && expressionNode instanceof BLangInvocation) {
```

Comment:
We can use `expressionNode.getKind() == NodeKind.INVOCATION` instead of `instanceof` check

Reply:
Fixed